src/Json_Encode.res

```diff
@@ -39,22 +39,22 @@ let option = encode => opt =>
   | Some(v) => v->encode
   }
 
-let withDefault = (default, encode, opt) =>
+let withDefault = (default, encode) => opt =>
   switch opt {
   | None => default
   | Some(v) => v->encode
   }
 
 let date = date => date->Js.Date.toJSONUnsafe->string
 
-let pair = (encodeA, encodeB, (a, b)) => [a->encodeA, b->encodeB]->jsonArray
-let tuple2 = (encodeA, encodeB, (a, b)) => [a->encodeA, b->encodeB]->jsonArray
-let tuple3 = (encodeA, encodeB, encodeC, (a, b, c)) =>
+let pair = (encodeA, encodeB) => ((a, b)) => [a->encodeA, b->encodeB]->jsonArray
+let tuple2 = (encodeA, encodeB) => ((a, b)) => [a->encodeA, b->encodeB]->jsonArray
+let tuple3 = (encodeA, encodeB, encodeC) => ((a, b, c)) =>
   [a->encodeA, b->encodeB, c->encodeC]->jsonArray
-let tuple4 = (encodeA, encodeB, encodeC, encodeD, (a, b, c, d)) =>
+let tuple4 = (encodeA, encodeB, encodeC, encodeD) => ((a, b, c, d)) =>
   [a->encodeA, b->encodeB, c->encodeC, d->encodeD]->jsonArray
 
-let dict = (encode, dict) => Js.Dict.map((. v) => encode(v), dict)->jsonDict
+let dict = encode => dict => Js.Dict.map(v => encode(v), dict)->jsonDict
```

Comment:
I appreciate the efforts in this PR, but this actually _forces_ uncurried mode. `Js.Dict.map` was hard-coded to an uncurried function in v11 😬 
https://github.com/rescript-lang/rescript-compiler/blame/2858ed7b86f28619ba7a3e44efa76973ba11ae39/jscomp/others/js_dict.resi#L173

Reply:
Hmm, not sure how to reconcile that. Are you gradually migrating to uncurried mode? Otherwise you could just keep using 1.3.

Reply:
I'm forking a library that depends on `bs-json` and had hoped to support both default and `uncurried:false` mode in v11, for people who are themselves migrating.

In the end some parts were so heavily tied to curried functions and the `bs-json` style that it can't support default v11 at all. At that point it was easier to just replace the JSON decoding with direct calls to `Js.Json` functions - it didn't use the more advanced features or even compose decoders together.